flair/trainers/trainer.py

```diff
@@ -674,13 +711,37 @@ def train(
                     print("saving model of current epoch")
                     model_name = "model_epoch_" + str(self.epoch) + ".pt"
                     self.model.save(base_path / model_name)
+                
+                if self.wandb_logger.wandb_run and log_model_each_k_epochs > 0 and not self.epoch % log_model_each_k_epochs:
+                    print("Logging model of current epoch")
+                    model_name = "model_epoch_" + str(self.epoch) + ".pt"
+                    if not Path(base_path / model_name).is_file():
+                        self.model.save(base_path / model_name)
+                    self.wandb_logger.log_artifact(
+                        name=f"model_{self.wandb_logger.wandb_run.id}",
+                        type="model",
+                        file_or_dir=base_path / model_name,
+                        aliases=[
+                            "latest",
+                            "best" if current_epoch_has_best_model_so_far else "",
+                            f"epoch-{self.epoch}",
+                        ],
+                    )
 
             if use_swa:
                 optimizer.swap_swa_sgd()
 
             # if we do not use dev data for model selection, save final model
             if save_final_model and not param_selection_mode:
                 self.model.save(base_path / "final-model.pt")
+                if self.wandb_logger.wandb_run:
+                    self.wandb_logger.log_artifact(
```

Comment:
I guess it uploads the model. It should be optional as model can weight many GBs.

Reply:
Oh yes. I missed this. I'll add another `log_final_model` flag here and lof the model only id its set. thanks for catching this

Reply:
@djstrong should be fixed now